force-app/main/default/classes/ErrorLogHandler.cls

```diff
@@ -0,0 +1,14 @@
+public class ErrorLogHandler {
+    public static void insertErrorLog( DateTime logDateTime, String logDetails, String processName) {
+        Error_Log__c errorLog = new Error_Log__c();
+        errorLog.Log_Date_Time__c = logDateTime;
+        errorLog.Log_Details__c = logDetails;
+        errorLog.Process_Name__c = processName;
+
+        try {
+            insert errorLog;
+        } catch (DmlException e) {
```

Comment:
If you catch only DmlException, you are bypassing other exceptions that may occur - for example Mixed DML exception or a LOCK ROW exception. You should catch the class "Exception" here instead so you will handle all errors.